XMonad/Prelude.hs

```diff
@@ -55,3 +60,11 @@ chunksOf i xs = chunk : chunksOf i rest
 -- > f .: g ≡ (f .) . g ≡ \c d -> f (g c d)
 (.:) :: (a -> b) -> (c -> d -> a) -> c -> d -> b
 (.:) = (.) . (.)
+
+-- | 'Data.List.NonEmpty.fromList' with a better error message. Useful to
+-- silence GHC's Pattern match(es) are non-exhaustive warning in places where
+-- the programmer knows it's always non-empty, but it's infeasible to express
+-- that in the type system.
+nonEmpty :: HasCallStack => [a] -> NonEmpty a
```

Comment:
This confused me when reading through the code top-to-bottom, so I'll mention it: `Data.List.NonEmpty.nonEmpty` already exists and returns a `Maybe`, making some of these view patterns really confusing to read if one has that function in mind

Reply:
Good point!

I wanted to change this to `notEmpty`, which looks a bit more like an assertion, which it is, but then forgot. Suggestions for better name welcome.

Reply:
fe79488cf0af04cf1631145c5268901be6a003e4